animations-tutorial.md

```diff
@@ -0,0 +1,639 @@
+
+We’ll use `addListener` in the next section and we’ll use `addStatusListener` in Monitoring the progress of the animation.
+
+## Rendering animations
+
+So far all we’ve done is to generate a sequence of numbers over time. You’ll notice we haven’t actually rendered anything.
+
+To render with an `Animation<>` object, store the `Animation<>` object as a member of your `Widget`, then use its value to decide how to draw.
+
+Consider this application that draws the Flutter logo without animation:
+
+```dart
+import 'package:flutter/material.dart';
+
+const kLogoUrl = 'https://raw.githubusercontent.com/dart-lang/logos/master/logos_and_wordmarks/dart-logo.png';
+
+class LogoApp extends StatefulComponent {
+  LogoAppState createState() => new LogoAppState();
+}
+
+class LogoAppState extends State<LogoApp> {
+
+  Widget build(BuildContext context) {
+    return new Center(
+        child: new Container(
+            margin: new EdgeDims.symmetric(vertical: 10.0),
+            height: 300.0,
+            width: 300.0,
+            child: new NetworkImage(src: kLogoUrl)));
+  }
+}
+
+void main() {
+  runApp(new LogoApp());
+}
+```
+
+This listing shows the changes necessary to animate the logo to
+grow outward: (Changes noted with comments.)
+
+```dart
+import 'package:flutter/animation.dart'; // New
+import 'package:flutter/material.dart';
+
+const kLogoUrl = "https://raw.githubusercontent.com/dart-lang/logos/master/logos_and_wordmarks/dart-logo.png";
+
+class LogoApp extends StatefulComponent {
+  LogoAppState createState() => new LogoAppState();
+}
+
+class LogoAppState extends State<LogoApp> {
+  Animation<double> animation;
+
+// Begin new
+
+  initState() {
+    super.initState();
+    AnimationController controller =
+        new AnimationController(duration: const Duration(milliseconds: 2000));
+    animation = new Tween(begin: 0.0, end: 300.0).animate(controller)
+      ..addListener(() {
+        setState(() {
+          // the state that has changed here is the animation object’s value
+        });
+      });
+    controller.forward();
+  }
+
+// End new
+
+  Widget build(BuildContext context) {
+    return new Center(
+        child: new Container(
+            margin: new EdgeDims.symmetric(vertical: 10.0),
+            height: animation.value,  // New
+            width: animation.value,   // New
+            child: new NetworkImage(src: kLogoUrl)));
+  }
+}
+
+void main() {
+  runApp(new LogoApp());
+}
+```
+
+Because `addListener` calls `setState`, every time the animation generates a new
+number, the current frame will be marked dirty, which in turn will force `build`
+to be called again. In `build`, the container changes size because its height
+and width now use `animation.value` instead of hardcoding 300.0.
+
+With these short changes, you’ve created your first animation in Flutter!
```

Comment:
I think that the Dart Language Tricks section was removed? The use of dot dot in examples was specifically identified by Adam as one of the stumbling blocks that people encountered in the codelab this week and it needs to be discussed. If you feel strongly that you don't want it in this doc, then please link another page or a stackoverflow post.